core/vm/intpool.go

```diff
@@ -65,3 +71,54 @@ func (p *intPool) put(is ...*big.Int) {
 		p.pool.push(i)
 	}
 }
+
+const poolPoolLimit = 5
+const (
+	PoolPoolTypeMiner = iota
+	PoolPoolTypeImporter
+	PoolPoolTypeTracer
+	PoolPoolTypeRPC
+	PoolPoolTypes
+)
+
+// IntPoolPool manages a pool of intPools
+type IntPoolPool struct {
+	pools [PoolPoolTypes][poolPoolLimit]*intPool
+
+	// Mutex
+	locks [PoolPoolTypes]sync.RWMutex
+}
+
+var intPoolPool = IntPoolPool{}
+
+// get is looking for an available pool to return
+func (ipp *IntPoolPool) get(poolType int) *intPool {
+	if poolType > PoolPoolTypes {
+		panic("Invalid intPool pool type requested")
+	}
+
+	ipp.locks[poolType].Lock()
+	defer ipp.locks[poolType].Unlock()
+
+	for i, ip := range ipp.pools[poolType] {
+		if ip.inUse == false {
+			ip.inUse = true
+			ip.poolIndex = i
+			return ip
+		}
+	}
+
+	panic("Could not find a pool")
```

Comment:
@karalabe What should happen if no extra pool can be allocated?

Reply:
You can always allocate a new pool if there's none pending. You only store the available ones, don't pre-allocate and do arrays. No need, simple slice. If it's empty, you make a new one.